### PR TITLE
Refactor model implementations

### PR DESCRIPTION
**Pull request by djdameln**
# Description
- Refactor the model implementations so that the pytorch modules and anomaly map generators are defined in separate files. 
- Fixes https://github.com/openvinotoolkit/anomalib/issues/211

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/models/cflow/torch_model.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+"""PyTorch model for CFlow model implementation."""
+
+# Copyright (C) 2020 Intel Corporation
```
Comment (ashwinvaidya17) [inline anomalib/models/cflow/torch_model.py:3]: Minor issue but should be be updating the dates? Is there a standard practice for this?

File: anomalib/models/padim/config_fldr.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+dataset:
```
Comment (ashwinvaidya17) [inline anomalib/models/padim/config_fldr.yaml:1]: Is this file an unintended commit?

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks! Only a few minor comments

File: anomalib/models/cflow/model.py [hunk 3]
```diff
@@ -17,215 +17,17 @@
 # See the License for the specific language governing permissions
```
Comment (samet-akcay) [inline anomalib/models/cflow/model.py:17]: I would personally prefer to rename `model.py` to `lightning_model.py` to match the pattern with `torch_model.py`

File: anomalib/models/__init__.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
 # TODO(AlexanderDokuchaev): Workaround of wrapping by NNCF.
 #                           Can't not wrap `spatial_softmax2d` if use import_module.
-from anomalib.models.padim.model import PadimLightning  # noqa: F401
+from anomalib.models.padim.lightning_model import PadimLightning  # noqa: F401
```
Comment (samet-akcay) [inline anomalib/models/__init__.py:28]: Is this import needed?

Comment (samet-akcay) [reply anomalib/models/__init__.py:28]: yeah looks like it. But I don't understand how/why it's needed. It is not used anywhere

**Pull request merged**